nodestream/pipeline/transformers/transformer.py

```diff
@@ -84,13 +84,16 @@ def drain_completed_tasks():
                 # Flush the pending tasks, then yield the flush.
                 # In order to fully respect the Flush,
                 # we need to wait for all pending tasks to complete.
+                # If we're still waiting we can yield the processor.
                 while pending_tasks:
                     for result in drain_completed_tasks():
                         yield result
+                    self.yield_processor()
```

Comment:
I wonder if the same result would be achieved by adding `await`:
```
for result in await drain_completed_tasks_async(pending_tasks):
```
I believe calling with an `await` will pass control back to the event loop.

Reply:
I believe making the function drain_completed_tasks_async turned it into an `AsyncGenerator` object which works on 
`async for result in drain_completed_tasks_async(pending_tasks):`